util/util.go

```diff
@@ -197,3 +200,33 @@ func VPCZonesForPowerVSRegion(region string) ([]string, error) {
 	}
 	return nil, fmt.Errorf("VPC zones corresponding to a PowerVS region %s not found ", region)
 }
+
+// IsGlobalRoutingRequiredForTG returns true when powervs and vpc regions are different.
+func IsGlobalRoutingRequiredForTG(powerVSRegion string, vpcRegion string) bool {
+	if r, ok := Regions[powerVSRegion]; ok && r.VPCRegion == vpcRegion {
+		return false
+	}
+	return true
+}
+
+// GetTransitGatewayLocationAndRouting returns appropriate location and routing suitable for transit gateway.
+// routing indicates whether to enable global routing on transit gateway or not.
+// returns true when PowerVS and VPC region are not same otherwise false.
+func GetTransitGatewayLocationAndRouting(powerVSZone *string, vpcRegion *string) (*string, *bool, error) {
+	if powerVSZone == nil {
+		return nil, nil, fmt.Errorf("powervs zone is not set")
+	}
+	powerVSRegion := endpoints.ConstructRegionFromZone(*powerVSZone)
+
+	if vpcRegion != nil {
+		routing := IsGlobalRoutingRequiredForTG(powerVSRegion, *vpcRegion)
+		return vpcRegion, &routing, nil
+	}
+	location, err := VPCRegionForPowerVSRegion(powerVSRegion)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to fetch vpc region associated with powervs region '%s': %w", powerVSRegion, err)
+	}
+
+	// returning routing as true since VPC region is not set and used PowerVS region to calculate the transit gateway location.
+	return &location, ptr.To(true), nil
```

Comment:
Just curious to know for setting true, This case may not occur as we have a webhook check for making sure the vpc region is set, but incase it occurs any how we are getting VPC region associated with PowerVS so can't we use local routing?

Reply:
thanks for pointing it out, its supposed to be local routing!😁